.github/workflows/build.yml

```diff
@@ -367,12 +370,12 @@ jobs:
     steps:
     - uses: actions/checkout@v2
       with:
-        ref: ${{ env.HEAD_SHA  }}
+        ref: ${{ inputs.sha  }}
     - id: push
       uses: ./.github/actions/push
       with:
-        head_sha: ${{ env.HEAD_SHA }}
-        artifact_name: image-${{ inputs.arch }}
+        head_sha: ${{ inputs.sha }}
```

Comment:
it looks like `push-pr` and `push-release` are identical with only the filter to enable being different, might be able to consolide to a single step with 
```
if: ${{ (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository) || (github.event_name == 'push' && github.ref == 'refs/heads/master') }} 
```

Reply:
They aren't identical. The push pr job runs after build before test, and the push release job runs after build and test. The main reason for this is tests can take a while to run. We want to be certain of releases before they are pushed on master, but on PRs it's more helpful to always have an image pushed even if it is broken.